test/DynamicDependency/Powershell/Test-PackageDependency.Tests.ps1

```diff
@@ -0,0 +1,285 @@
+# Copyright (c) Microsoft Corporation and Contributors.
+# Licensed under the MIT License.
+
+Set-StrictMode -Version 3.0
+
+$ErrorActionPreference = "Stop"
+
+# Repository paths
+$root = (Get-Item $PSScriptRoot).parent.parent.parent.FullName
+$dev = Join-Path $root 'dev'
+$test = Join-Path $root 'test'
+$buildOutput = Join-Path $root 'BuildOutput\Debug\x64'
+$cmdlet = Join-Path $dev 'DynamicDependency\Powershell'
+
+# Test package(s)
+$packageName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets'
+$packageFamilyName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets_8wekyb3d8bbwe'
+$packageFullName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets_1.2.3.4_neutral__8wekyb3d8bbwe'
+$packageMsixDir = 'Framework.Widgets'
+$packageMsixFilename = 'Framework.Widgets.msix'
+$packageMsixPath = Join-Path $buildOutput $packageMsixDir
+$packageMsix = Join-Path $packageMsixPath $packageMsixFilename
+
+# Track resources to ensure we clean up
+$packageDependencyIds = [ordered]@{}
+$packageDependencyContexts = [ordered]@{}
+
+Function LogComment
+{
+    param([string]$message)
+
+    "       $message" | Write-Host
+}
+
+Function Trace
+{
+    param([string]$message)
+
+    Write-Host $message -ForegroundColor Magenta
+}
+
+Function TraceEnter
+{
+    param([string]$message)
+
+    Trace " Test: $message"
+}
+
+Function RemoveTestPackages
+{
+    Write-Host "Removing $packageName..." -ForegroundColor Cyan
+
+    # Remove our package(s) in case they were previously installed and incompletely removed
+    Get-AppxPackage $packageName | Remove-AppxPackage
+    $p = Get-AppxPackage $packageName
+    if ($p -ne $null)
+    {
+        Write-Error $p
+        Write-Error "Remove/Get-AppxPackage result not expected"
+        Exit 1
+    }
+}
+
+Function AddTestPackages
+{
+    Write-Host "Adding $packageMsix..." -ForegroundColor Cyan
+
+    # Install our needed package(s)
+    if (-not(Test-Path -Path $packageMsix -PathType Leaf))
+    {
+        Write-Error "$($packageMsix) not found"
+        Exit 1
+    }
+    Add-AppxPackage $packageMsix
+    $p = Get-AppxPackage $packageName
+    if (($p -eq $null) -Or ($p.PackageFullName -ne $packageFullName))
+    {
+        Write-Error $p
+        Write-Error "Get-AppxPackage result not expected"
+        Exit 1
+    }
+    if ($p.PackageFullName -ne $packageFullName)
```

Comment:
If this condition is true we would have exited via the previous if-block?

Reply:
No. They test different things. 

line 69 tests if a file exists

```
if (-not(Test-Path -Path $packageMsix -PathType Leaf))
```

line 76 tests if a package is registered

```
    Add-AppxPackage $packageMsix
    $p = Get-AppxPackage $packageName
    if (($p -eq $null) -Or ($p.PackageFullName -ne $packageFullName))
```

When run line 66 shows $packageMsix is a .msix file

```
Adding D:\source\repos\windowsappsdk2\BuildOutput\Debug\x64\Framework.Widgets\Framework.Widgets.msix...
```